### PR TITLE
qt59: darwin compatibility

### PR DESCRIPTION
**Pull request by acowley**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by acowley** [comment]
Ping @j-hao : can you take a look at this? The `sed` patches are hacky, but are enough to convince the compiler.

**Comment by acowley** [comment]
I've moved the `-qt-harfbuz` flag into a darwin guarded section where it should have been. These fixes shouldn't affect linux.

File: pkgs/development/libraries/qt-5/modules/qtbase.nix [hunk 1]
```diff
@@ -135,10 +135,15 @@ stdenv.mkDerivation {
...
-          substituteInPlace ./mkspecs/common/mac.conf \
-              --replace "/System/Library/Frameworks/AGL.framework/" "${darwin.apple_sdk.frameworks.AGL}/Library/Frameworks/AGL.framework/"
+      substituteInPlace ./mkspecs/common/mac.conf \
+        --replace "/System/Library/Frameworks/OpenGL.framework/" "${darwin.apple_sdk.frameworks.OpenGL}/Library/Frameworks/OpenGL.framework/"
+      substituteInPlace ./mkspecs/common/mac.conf \
+        --replace "/System/Library/Frameworks/AGL.framework/" "${darwin.apple_sdk.frameworks.AGL}/Library/Frameworks/AGL.framework/"
+
+      # Recent Qt releases use SDK (and Objective-C) features beyond
+      # the 10.10 available in nixpkgs
+      sed 's/NSEnumerator<NSWindow\*>/NSEnumerator/' -i ./src/plugins/platforms/cocoa/qcocoawindow.mm
```
Comment (LnL7) [inline pkgs/development/libraries/qt-5/modules/qtbase.nix]: I prefer using substituteInPlace for literals so you don't have to worry about escaping stuff.

**Review by LnL7 — APPROVED** [review]

**Comment by LnL7** [comment]
@GrahamcOfBorg build qt59.qtbase

**Comment by LnL7** [comment]
@grahamc hmm, evaluates fine on hydra.

**Comment by LnL7** [comment]
@GrahamcOfBorg build qt59.qtbase

**Comment by acowley** [comment]
Okay, I tend to have more trouble with `substituteInPlace` than `sed` and it seems like another way of doing the same thing, but if someone can rewrite those `sed` replacements, I'm happy to change them out. I tried copying the template suggested by the earlier `substituteInPlace` calls, but they ended up erasing the contents of the files.

**Comment by acowley** [comment]
I've swapped out the uses of `sed` for `substituteInPlace`. It fails for me in `nix-shell`, but the `nix-build` worked. This is apparently due to something wrong on my systems.

**Comment by acowley** [comment]
It looks like @ttuegel fixed the qt issues independently. @LnL would you like to separate your `gstreamer` fix, and I'll just close this PR?

**Pull request closed without merging**